src/codegen/statements/try_catch.rs

```diff
@@ -459,31 +512,17 @@ fn insert_catch_clauses(
         kind: Builtin::ReadFromBuffer,
         args: vec![buffer.clone(), offset],
     };
-
-    cfg.set_basic_block(enough_data_block);
-    // Expect the returned data to match the 4 bytes function selector for "Error(string)"
-    let err_id = Expression::NumberLiteral {
-        loc: Codegen,
-        ty: Type::Bytes(4),
-        value: ERROR_SELECTOR.into(),
-    };
-    let cond = Expression::Equal {
-        loc: Codegen,
-        left: err_selector.into(),
-        right: err_id.into(),
-    };
-    let match_err_id = cfg.new_basic_block("match_err_id".into());
-    let instruction = Instr::BranchCond {
-        cond,
-        true_block: match_err_id,
-        false_block: no_match_err_id,
+    let instruction = Instr::Switch {
```

Comment:
The switch looks very nice here, great! :heart_eyes:

Reply:
Thanks!